js/job.jsx

```diff
@@ -75,15 +73,9 @@ export default class Job extends React.Component {
         ),
       })
     ));
-  }
 
-  componentWillReceiveProps(next) {
```

Comment:
Removing this will break the case of navigating from job detail page A to job detail page B via a link on A. When that happens, the Job component and page structure stays the same but we get a new jobId, and when the jobid changes we need to kick off a fetch to go get the data for job B. 

Removing `componentWillReceiveProps` on the other components is less obviously broken since there are not links between counter sub-pages for different jobs. However, it will break changing the URL via the URL bar since the jobid will change but new data will not be fetched.

Reply:
You probably want to factor all the we-have-a-new-jobid state changes out to a method that is called in both DidMount and WillReceiveProps.

Reply:
done!